packages/runtime/container-runtime/src/gc/gcTelemetry.ts

```diff
@@ -109,11 +91,16 @@ export class GCTelemetryTracker {
 		private readonly getNodeStateTracker: (
 			nodeId: string,
 		) => UnreferencedStateTracker | undefined,
-		private readonly getNodePackagePath: (
-			nodePath: string,
-		) => Promise<readonly string[] | undefined>,
 	) {}
 
+	/**
+	 * Called after every GC run. Clear all references since the last GC run since the unreferenced
+	 * state of all nodes would have updated and the references are now stale.
+	 */
+	public gcRunCompleted() {
+		this.referencesSinceLastRun.clear();
```

Comment:
How about just passing in `newReferencesSinceLastRun` from GC into `nodeUsed`, instead of tracking this thing?  Seems likely to be redundant.

Reply:
That is what I started with. But `newReferencesSinceLastRun` is not a list of references. It is a list of `nodeId -> references`. So, we will have to get the references from it every time `nodeUsed` is called which is for every op making it very expensive.